dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,131 @@
+            return true
+        }
+        return false
+    }
+
+    private fun verifyDataIdIsRegistered(dataId: String): Boolean {
+        if (dataMetaData.containsKey(dataId)) {
+            return true
+        }
+        return false
+    }
+
+    /*
+    ________________________________
+    Methods to route data inserts and queries to the data store and save meta data in the Dataland-Meta-Data-Storage:
+    ________________________________
+     */
+
+    override fun addDataSet(storedDataSet: StorableDataSet): String {
+        if (!verifyCompanyIdExists(storedDataSet.companyId)) {
+            throw IllegalArgumentException("The companyId: ${storedDataSet.companyId} does not exist.")
+        }
+        val dataId = dataStore.insertDataSet(storedDataSet.data)
+        this.dataMetaData[dataId] =
+            DataSetMetaInformation(dataType = storedDataSet.dataType, companyId = storedDataSet.companyId)
+        this.companyData[storedDataSet.companyId]!!.dataSets.add(
+            DataIdentifier(dataId = dataId, dataType = storedDataSet.dataType)
+        )
+        return dataId
+    }
+
+    override fun getDataSet(dataIdentifier: DataIdentifier): String {
+        if (!verifyDataIdIsRegistered(dataIdentifier.dataId)) {
+            throw IllegalArgumentException("Dataland does not know a data set with the id: ${dataIdentifier.dataId} ")
+        }
+
+        if (dataStore.selectDataSet(dataIdentifier.dataId) == "") {
+            throw IllegalArgumentException(
+                "No data set with the id: ${dataIdentifier.dataId} " +
+                    "could be found in the data store."
+            )
+        }
+        if (dataMetaData[dataIdentifier.dataId]!!.dataType != dataIdentifier.dataType) {
+            throw IllegalArgumentException(
+                "The data with the id: ${dataIdentifier.dataId} is registered as type" +
+                    " ${dataMetaData[dataIdentifier.dataId]} by Dataland instead of your requested" +
+                    " type ${dataIdentifier.dataType}."
+            )
+        }
+        return dataStore.selectDataSet(dataIdentifier.dataId)
+    }
+
+    /*
+    ________________________________
+    Methods to process meta data:
+    ________________________________
+     */
+
+    override fun getMetaData(dataId: String): DataSetMetaInformation {
+        if (!verifyDataIdIsRegistered(dataId)) {
+            throw IllegalArgumentException("Dataland does not know a data set with the id: $dataId ")
+        }
+        return dataMetaData[dataId]!!
+    }
+
+    /*
+    ________________________________
+    Methods to add and retrieve company info and save associated company meta data in the
+    Dataland-Company-Meta-Data-Storage:
+    ________________________________
+     */
+
+    override fun addCompany(companyName: String): CompanyMetaInformation {
+        companyCounter++
+        companyData["$companyCounter"] = StorableCompany(companyName = companyName, dataSets = mutableListOf())
```

Comment:
Wiese als String? bekommen wir kein Problem wenn wir dier sortieren/auflisten/usw möchten?

Das mir immer noch nicht klar wieso haben wir ein Map

Reply:
Wieso Map:

![image](https://user-images.githubusercontent.com/99255296/160084702-95544532-a111-4a74-9450-0ced91248256.png)

Reply:
Ich denke IDs könnten auch Integer sein. Wäre kein Problem. Ich habe es bisher ehrlich gesagt im Code noch nicht hinterfragt.
Einzige Sache die mir in den Sinn kommt:
In der Vergangenheit hatte ich Fälle, wo IDs mit "0" anfangen.  Diese können nicht korrekt als Integers behandelt werden.
Wenn wir jetzt noch nicht wissen, welche Art IDs von EuroDaT zurückkommen, ist String erstmal safer.

Was meinst du @SiegfriedSobkowiak ?

Reply:
Heute nochmals besprochen mit Siegfried.   Wir bleiben bei Maps